galsim/phase_psf.py

```diff
@@ -0,0 +1,1442 @@
+    significantly aberrated optical PSFs without atmospheric aberrations -- it may be useful to
+    further increase the size of the pupil plane array, thereby increasing the sampling rate of the
+    resulting PSF image.  This can be done by increasing the `oversampling` keyword.
+
+    The pupil plane array physical sampling interval (which is directly related to the resulting PSF
+    image physical size) is set by default to the same interval as would be used to avoid
+    significant aliasing (image folding) for an obscured Airy profile with matching diameter and
+    obscuration and for the value of `folding_threshold` in the optionally specified gsparams
+    argument.  If the phase aberrations are significant, however, the PSF image size computed this
+    way may still not be sufficiently large to avoid aliasing.  To further increase the pupil plane
+    sampling rate (and hence the PSF image size), you can increase the value of the `pad_factor`
+    keyword.  An additional way to set the pupil sampling interval for a particular set of phase
+    screens (i.e., for a particular PhaseScreenList) is to provide the screens in the `screen_list`
+    argument.  Each screen in the list computes its own preferred sampling rate and the
+    PhaseScreenList appropriately aggregates these. This last option also requires that a wavelength
+    `lam` be specified, and is particularly helpful for creating PSFs derived from turbulent
+    atmospheric screens.
+
+    Finally, when specifying the pupil geometrically, Aperture may choose to make a small adjustment
+    to `pupil_plane_scale` in order to produce an array with a good size for FFTs.  If your
+    application depends on knowing the size and scale used with the Fourier optics framework, you
+    can obtain these from the `Aperture.pupil_plane_size` and `Aperture.pupil_plane_scale`
+    attributes.
+
+    Pupil image specification
+    --------------------------
+
+    The second way to specify the pupil plane configuration is by passing in an image of it.  This
+    can be useful, for example, if the struts are not evenly spaced or are not radially directed, as
+    is assumed by the simple model for struts described above.  In this case, an exception is raised
+    if keywords related to struts are also given.  On the other hand, the `obscuration` keyword is
+    still used to ensure that the PSF images are not aliased, though it is ignored during the actual
+    construction of the pupil plane illumination pattern.  The `pupil_plane_im` that is passed in
+    can be rotated during internal calculations by specifying a `pupil_angle` keyword.  Also note
+    that given how the `obscuration` keyword is used, the obscuration in the supplied pupil plane
+    image should indeed be reasonably circular rather than having an arbitrary shape; however, minor
+    deviations from a perfect circle do not cause significant problems.
```

Comment:
I'm uncomfortable with this last sentence.  It seems like a "should" that the user does not actually have any control over.  It's not up to the user how the telescope obscuration looks, so there's nothing they can do to fix this if the obscuration is highly non-circular.  

Furthermore, I think the struts actually cause significantly more problems for aliasing than a non-circular obscuration.  We've actually been fairly lax with the aliasing calculation in the presence of struts.  They properly should get smaller stepk values than we actually use, but we've ignored that in the spirit of "realistically complex" but not perfectly precise PSFs.

In any case, for this paragraph, I'd recommend just ditching this last sentence.  Or possibly saying that if the spider pattern is very complex, then the user might want to increase pad_factor for more fidelity at the expense of slower running time.